scheduler/gform_backend.py

```diff
@@ -0,0 +1,395 @@
+                "car_type": row[CAR_DESCRIPTION_COLUMN],
+                "seats": int(row[SEATS_COLUMN]),
+                "is_dues_paying": True,
+                "is_driver": True,
+                "days": [],
+            }
+
+            # assume each day has a locations column and a departure times column
+            driver_days_start = DAYS_INFO_START_COLUMN + 2 * len(days_enabled)
+
+            # create a dict for each day
+            for (i, d) in zip(
+                range(driver_days_start, driver_days_start + len(days_enabled)),
+                days_enabled,
+            ):
+                day = dict()
+
+                # if driving this day
+                if row[i]:
+
+                    day["day"] = d
+
+                    # add locations
+                    day["locations"] = []
+                    locations = row[i].split(",")
+
+                    for l in locations:
+                        day["locations"].append(parse_location(l.strip()))
+
+                    # driver only has one departure time
+                    day["departure_times"] = [
+                        parse_times(row[i + len(days_enabled)].split(",")[0])
+                    ]
+
+                    driver["days"].append(day)
+
+            drivers.append(driver)
+
+    return drivers
+
+
+def members_from_sheet(dues_payers, responses, days_enabled):
+    """
+    Gets all club members who submitted a response using the form.
+    """
+
+    client = AuthorizedClient.get_instance().client
+
+    for sheet in client.openall():
+        logger.debug("%s", sheet.title)
+
+    # get responses and dues payers sheets
+    responses_sheet = client.open(responses).sheet1
+    dues_payers_sheet = client.open(dues_payers).sheet1
+
+    all_responses = responses_sheet.get_all_values()
+
+    # create lists of riders and drivers
+    riders = get_riders(all_responses, days_enabled, get_dues_payers(dues_payers_sheet))
+    drivers = get_drivers(days_enabled, all_responses)
+
+    return riders, drivers
+
+
+def delete_spreadsheet(names):
+    """
+    Deletes the spreadsheets in names
+    """
+
+    for n in names:
+        n.strip()
+
+    client = AuthorizedClient.get_instance().client
+
+    for s in client.openall():
+        if s.title in names:
+            client.del_spreadsheet(s.id)
+
+
+def create_spreadsheet(name, location, client):
+    """
+    Creates a spreadsheet with given name in the location defined by the folder_id.
+    The folder_id can be found by viewing the folder in Drive and selecting the 
+    id
+    
+    drive.google.com/../folders/1/<id>
+    """
+
+    # checks if sheet with name already exists, deletes if exists
+    if client.openall(name):
+        delete_spreadsheet(name)
+        logger.info("Sheet exists, deleting")
+
+    logger.info("Creating sheet")
+    spreadsheet = client.create(name, folder_id=location)
+    spreadsheet.share(
+        "rkalnins@umich.edu", notify=False, perm_type="user", role="writer"
+    )
```

Comment:
If you're running this you might need to share with you account as well to view. not sure. it might be shared as long as it's in the folder